src/queue.hpp

```diff
@@ -729,11 +777,47 @@ struct cvk_command_batch : public cvk_command {
 
     cl_uint batch_size() { return m_commands.size(); }
 
-    bool is_profiled_by_executor() const override final { return false; }
+    CHECK_RETURN cl_int set_profiling_info(cl_profiling_info pinfo,
+                                           cl_int status) override final {
+        const cl_int target_status = status;
+        status = cvk_command::set_profiling_info(pinfo, status);
+        if (m_queue->profiling_on_device() &&
+            pinfo == CL_PROFILING_COMMAND_START) {
+            if (m_queue->device()->has_timer_support()) {
+                cl_ulong sync_dev, sync_host;
+                auto ret = m_queue->device()->get_device_host_timer(&sync_dev,
+                                                                    &sync_host);
+                if (ret != CL_SUCCESS)
+                    return ret;
+                for (auto& cmd : m_commands) {
+                    cmd->set_sync_values(sync_dev, sync_host);
+                }
```

Comment:
I wanted to avoid calling `vkGetCalibratedTimestampsEXT` for each command of the batch.
So it is called only once in the batch command, and then copied to all the commands in the batch.

Reply:
I understood that you were trying to not run it for each command in the batch (which is definitely good!) but I was wondering whether we really needed to store the values in each command. We could probably pass the values from the batch when they're needed. I'll have a go at refactoring this and see if it looks better.